sgt-back-end/server.js

```diff
@@ -0,0 +1,232 @@
+import express from 'express';
+import ClientError from './client-error.js';
+import errorMiddleware from './error-middleware.js';
+import pg from 'pg';
+
+const port = 8080;
+const app = express();
+const parser = express.json();
+app.use(parser);
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+/**
+ * Checks whether the provided score is INVALID: meaning not in our range or not an integer.
+ * Returns boolean, TRUE if is not valid, FALSE if it is valid.
+ */
+function checkInvalidScore(score) {
```

Comment:
These can be even nicer now that we're using exceptions. This is what I would do:
```
function validateScore(score) {
  score = Number(score);
  if (score > 100 || score < 0 || !Number.isInteger(score)) {
    throw new ClientError(400, 'Invalid score');
  }
}
```
Then you can just call `validateScore` in your handler and not have to worry about checking its result and throwing.